lib/restrictions.js

```diff
@@ -273,19 +289,16 @@ async function _limitOverDuration(
   {/*acquirerId,*/ acquired, request, /*zones,*/ restriction}) {
   const {methodOptions: {limit, duration}} = restriction;
 
-  // TODO: fully support ISO 8601 `duration` w/another library, this only
-  // handles a duration of a period of 30 days
-  if(duration !== 'P30D') {
-    // bad data of this sort should be prevented by validators such that
-    // this code path isn't hit
-    throw new Error(`Unsupported duration "${duration}".`);
-  }
+  // parse duration
+  const durationObject = parse(duration);
 
   // determine when the duration started
-  const now = new Date();
-  const start = new Date();
-  start.setDate(now.getDate() - 30);
-  const startTime = start.getTime();
+  let startTime = new Date();
+  for(const timeUnitKey in durationObject) {
```

Comment:
it appears to me that `iso8601-duration` provides an `end` API that can be used here:

https://github.com/tolu/ISO8601-duration/blob/master/lib/index.js#L58

Reply:
The `end` API gives the duration in the future date (so if the duration is P30D it will give a date 30 days in future). We are looking for the previous date, so in the P30D example we are looking for 30 days ago.

Reply:
Yes, I see that now after studying the algo more carefully.  I was imagining a case where some "acquisitionExiprationDate" was computed from `acquistion.requested (date)` and then one would simply compare determine if that date was LTE now().  Looks to me like the current algorithm is more efficient than this idea.  Marking this resolved.  TY